### PR TITLE
[Buttons] Update name for fallback behavior.

### PR DESCRIPTION
**Pull request by codeman7**
## Motivation
As a team we have decided that `mdc_legacyFontScaling` isn't the correct name for what the flag actually does. We have decided to change the name to `adjustFontForContentSizeCategoryWhenScaledFontIsUnavailable` is a more accurate name.

## Detailed changes
- Deprecate the older API `mdc_legacyFontScaling` so we can delete it in the upcoming release.
- Add API for `adjustFontForContentSizeCategoryWhenScaledFontIsUnavailable`
- Update the getter and setter for `mdc_legacyFontScaling` to mirror `adjustFontForContentSizeCategoryWhenScaledFontIsUnavailable`

### CODE REVIEW

**Review by romoore — COMMENTED** [review]
Please add tests for the new API.

File: components/Buttons/tests/unit/ButtonsTests.m [hunk 1]
```diff
@@ -1116,6 +1116,19 @@ - (void)testAdjustsFontProperty {
...
+}
+
+- (void)testAdjustsFontForContentSizeCategoryWhenScaledFontIsUnavailableSetsValueCorrectly {
+  // When
+  self.button.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
+
+  // Then
+  XCTAssertFalse(self.button.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable);
+}
+
```
Comment (jverkoey) [inline components/Buttons/tests/unit/ButtonsTests.m:1161]: Can we add a test that tests the expected code path for adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable? E.g. two separate tests, one for the false behavior, one for the true behavior, that validates that the fallback behavior works as expected?

Comment (codeman7) [reply components/Buttons/tests/unit/ButtonsTests.m:1161]: Snapshot test have been added, these changes are all reverted. The snapshot test cover when `adjustFont....` is both `YES` and `NO`.

File: components/Buttons/tests/unit/ButtonsTests.m [hunk 2]
```diff
@@ -1116,6 +1120,45 @@ - (void)testAdjustsFontProperty {
...
+  // When
+  self.button.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
+  [self.button updateTitleFont];
+
+  // Given
+  XCTAssertTrue([self.button.titleLabel.font mdc_isSimplyEqual:originalFont]);
+}
+
+/**
+ Test legacy dynamic type has no impact when @c
```
Comment (jverkoey) [inline components/Buttons/tests/unit/ButtonsTests.m]: This comment appears to be out of date; Dynamic Type does have an impact here.

**Review by jverkoey — APPROVED** [review]

File: components/Buttons/tests/unit/ButtonsTests.m (hunk 2 above)
Comment (codeman7) [reply components/Buttons/tests/unit/ButtonsTests.m]: Whoops, copy and paste got me!

**Pull request merged**